test/common/test_point_type_construction.cpp

```diff
@@ -0,0 +1,804 @@
+  static_assert(pt.data[3] == 1.0f, "");
+}
+
+TEST (PointTypeConstruction, PointXYZThreeScalarsConstruction)
+{
+  constexpr const pcl::PointXYZ pt(2.0f, 3.0f, 4.0f);
+  static_assert(pt.data[0] == 2.0f, "");
+  static_assert(pt.data[1] == 3.0f, "");
+  static_assert(pt.data[2] == 4.0f, "");
+  static_assert(pt.data[3] == 1.0f, "");
+}
+
+TEST (PointTypeConstruction, PointXYZIDefaultConstruction)
+{
+  constexpr const pcl::PointXYZI pt;
+  static_assert(pt.data[0] == 0.0f, "");
+  static_assert(pt.data[1] == 0.0f, "");
+  static_assert(pt.data[2] == 0.0f, "");
+  static_assert(pt.data[3] == 1.0f, "");
+  static_assert(pt.intensity == 0.0f, "");
+}
+
+TEST (PointTypeConstruction, PointXYZIFourScalarsConstruction)
+{
+  constexpr const pcl::PointXYZI pt(2.0f, 3.0f, 4.0f, 5.0f);
+  static_assert(pt.data[0] == 2.0f, "");
+  static_assert(pt.data[1] == 3.0f, "");
+  static_assert(pt.data[2] == 4.0f, "");
+  static_assert(pt.data[3] == 1.0f, "");
+  static_assert(pt.intensity == 5.0f, "");
+}
+
+TEST (PointTypeConstruction, PointXYZLDefaultConstruction)
+{
+  constexpr const pcl::PointXYZL pt;
+  static_assert(pt.data[0] == 0.0f, "");
+  static_assert(pt.data[1] == 0.0f, "");
+  static_assert(pt.data[2] == 0.0f, "");
+  static_assert(pt.data[3] == 1.0f, "");
+  static_assert(pt.label == 0.0f, "");
+}
+
+TEST (PointTypeConstruction, PointXYZLFourScalarsConstruction)
+{
+  constexpr const pcl::PointXYZL pt(2.0f, 3.0f, 4.0f, 5.0f);
+  static_assert(pt.data[0] == 2.0f, "");
+  static_assert(pt.data[1] == 3.0f, "");
+  static_assert(pt.data[2] == 4.0f, "");
+  static_assert(pt.data[3] == 1.0f, "");
+  static_assert(pt.label == 5.0f, "");
+}
+
+TEST (PointTypeConstruction, IntensityDefaultConstruction)
+{
+  constexpr const pcl::Intensity pt;
+  static_assert(pt.intensity == 0.0f, "");
+}
+
+TEST (PointTypeConstruction, IntensityOneScalarConstruction)
+{
+  constexpr const pcl::Intensity pt(1.0f);
+  static_assert(pt.intensity == 1.0f, "");
+}
+
+TEST (PointTypeConstruction, Intensity8uDefaultConstruction)
+{
+  constexpr const pcl::Intensity8u pt;
+  static_assert(pt.intensity == std::uint8_t{}, "");
+}
+
+TEST (PointTypeConstruction, Intensity8uOneScalarConstruction)
+{
+  constexpr const pcl::Intensity8u pt(1u);
+  static_assert(pt.intensity == std::uint8_t{1u}, "");
+}
+
+TEST (PointTypeConstruction, Intensity32uDefaultConstruction)
+{
+  constexpr const pcl::Intensity8u pt;
+  static_assert(pt.intensity == std::uint32_t{}, "");
+}
+
+TEST (PointTypeConstruction, Intensity32uOneScalarConstruction)
+{
+  constexpr const pcl::Intensity32u pt(1u);
+  static_assert(pt.intensity == std::uint32_t{1u}, "");
+}
+
+using AllPointTypes = ::testing::Types<pcl::PointXYZ, pcl::RGB, pcl::Intensity, pcl::PointXYZI>;
+TYPED_TEST_SUITE (PointTypeConstexprConstructorTest, AllPointTypes);
+
+TYPED_TEST (PointTypeConstexprConstructorTest, ConstexprDefaultConstructionTests)
+{
+  static_assert(std::is_default_constructible<TypeParam>::value, "");
+}
```

Comment:
Don't intersect PointTypeConstruction tests with this

Reply:
i removed it, don't think it makes sense here, since there's no logic based on the fact that the point type is default constructible

Reply:
You can just add all the PointTypeConstruction tests and then the PointTypeConstexprConstructorTest

Reply:
i feel like we mix orthogonal assumptions here
if i understand correctly: 
1. there should be the same number of tests that test construction of point types in non-constexpr (runtime) context
2. there should be no tests with is_default_constructible trait in action
am i right?

Reply:
added additional runtime tests with gtest assertions testing float comparisons